documentation/faq/70-packages.md

```diff
@@ -2,10 +2,14 @@
 question: How do I fix the error I'm getting trying to include a package?
 ---
 
-SSR in Vite is not yet stable. Libraries work best with Vite when they distribute both CJS and ESM in their package and you may wish to work with library authors to make this happen.
+Vite's SSR support has become fairly stable since Vite 2.7. Most issues related to including a library are due to incorrect packaging.
 
-Svelte components must be written entirely in ESM. It is encouraged to make sure the dependencies of external Svelte components provide an ESM version. However, in order to handle CJS dependencies [`vite-plugin-svelte` will look for any CJS dependencies of external Svelte components](https://github.com/sveltejs/vite-plugin-svelte/blob/main/docs/faq.md#what-is-going-on-with-vite-and-pre-bundling-dependencies) and ask Vite to pre-bundle them by automatically adding them to Vite's `optimizeDeps.include` which will use `esbuild` to convert them to ESM.
+Libraries work best with Vite when they distribute an ESM version and you may wish to suggest this library authors. Here's a few things to keep in mind when checking if a library is packaged correctly:
+- `exports` takes precedence over the other fields like `main`, `module`, and `browser`. Adding an `exports` field may not be backwards-compatible as it prevents deep imports.
+- `main` should be defined if `exports` is not. It should be a CommonJS file unless `"type": "module"` is set, in which case it should be an ESM file. If a `module` field is defined, it should refer to an ESM file.
```

Comment:
If `"type": "module"` is set, should `main` always be an ESM file? I think it can be CJS too, depending on the library output. But either ways, both would prevent the opposite format from loading, which is probably intentional by Node.

The only way for dual format packages per [node](https://nodejs.org/api/packages.html#dual-commonjses-module-packages) seems to rely on `exports`

Reply:
I think older node doesn't know about exports, so you need main to point to the cjs file to create a hybrid package.

The even more fun part is that if you leave the main attribute populated and then try to use the package with Sapper (SSR), it won't work, even on modern node. So I found myself deleting the main field entirely.

Reply:
Yeah, this was worded slightly more restrictive than actually necessary. I've updated it so that it should be accurate now